src/chat/chat.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Chat } from './entity/chat.model';
+import { Repository } from 'typeorm';
+import { CreateChatDto, UpdateChatDto } from './dto/chat.dto';
+import { ERROR_MESSAGES } from '@app/common/constants/errors/error.messages';
+
+@Injectable()
+export class ChatService {
+  constructor(
+    @InjectRepository(Chat)
+    private chatRepository: Repository<Chat>,
+  ) {}
+
+  async createChat(chat: CreateChatDto) {
+    try {
+      const newChat = new Chat();
+
+      newChat.customer_id = chat.customer_id;
+      newChat.start_time = new Date();
+      newChat.status = 'open';

```

Comment:
change `status` to boolean since its only have 2 states, also change its name to `isOpen`